src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import duke.commands.*;
+
+public class Parser {
+
+    public static Command parse(String input) throws DukeException {
+        String arr1[] = input.split("/");
+        String arr2[] = arr1[0].split(" ");
+        switch (arr2[0]) {
+            case "list":
+                return new ListCommand();
+
+            case "mark":
+                if (arr2.length < 2) {
+                    throw new DukeException("OOPS!!! The description of a mark cannot be empty.");
+                }
+                int index = Integer.parseInt(arr2[1]);
+                return new MarkCommand(index);
+
+            case "unmark":
+                if (arr2.length < 2) {
+                    throw new DukeException("OOPS!!! The description of a unmark cannot be empty.");
+                }
+                int No = Integer.parseInt(arr2[1]);
```

Comment:
Shouldn't this variable be named in camelCase? i.e. `no`

Maybe you could consider renaming this variable to something like `index` to make its purpose clearer.